mirrord-layer/src/pod_api.rs

```diff
@@ -120,6 +122,16 @@ pub async fn create_agent(
         .unwrap();
 
     let pods_api: Api<Pod> = Api::namespaced(client.clone(), agent_namespace);
+    let params = ListParams::default().labels(&format!("job-name={}", agent_job_name));
+    let mut stream = pods_api.watch(&params, "0").await.unwrap().boxed();
+    while let Some(status) = stream.try_next().await.unwrap() {
```

Comment:
need to add timeout on that

Reply:
not sure but what if the resource has not been added yet and if we add a timeout, wouldnt it break out of the loop without the pod being there? the docs say the default timeout is 295 seconds though

Reply:
well, yes but we don't want it to wait indefinitely

Reply:
added a 10 second timeout